nixos/modules/config/pulseaudio.nix

```diff
@@ -254,24 +254,35 @@ in {
       };
     })
 
-    (mkIf systemWide {
+    (mkIf systemWide (let
+      cookiePath = "${stateDir}/.config/pulse/cookie";
+    in {
       users.extraUsers.pulse = {
         # For some reason, PulseAudio wants UID == GID.
         uid = assert uid == gid; uid;
         group = "pulse";
         extraGroups = [ "audio" ];
         description = "PulseAudio system service user";
         home = stateDir;
-        createHome = true;
       };
 
+      users.groups.pulse-access = {};
+
+      # Pulseaudio creates auth files like .esd_auth and cookie in user
+      # read-only mode, but then users are not able to use pulseaudio.
+      system.activationScripts.pulse = ''
+        umask 027
+        mkdir -p $(dirname ${cookiePath})
+        touch "${cookiePath}" "${stateDir}/.esd_auth"
+        chown -R pulse:pulse-access "${stateDir}"
+      '';
+
       users.extraGroups.pulse.gid = gid;
 
       systemd.services.pulseaudio = {
         description = "PulseAudio System-Wide Server";
         wantedBy = [ "sound.target" ];
         before = [ "sound.target" ];
-        environment.PULSE_RUNTIME_PATH = stateDir;
```

Comment:
Why do you delete this?

Reply:
Default location is the users home directory, which points to stateDir. This line is therefor unnecessary and causes more confusion instead of clarifications.

Reply:
OK.